tools/execsnoop.py

```diff
@@ -193,19 +197,19 @@ def print_event(cpu, data, size):
     elif event.type == EventType.EVENT_RET:
         if args.fails and event.retval == 0:
             skip = True
-        if args.name and not re.search(args.name, event.comm):
+        if args.name and not re.search(bytes(args.name), event.comm):
             skip = True
-        if args.line and not re.search(args.line,
-                                       b' '.join(argv[event.pid]).decode()):
+        if args.line and not re.search(bytes(args.line),
+                                       b' '.join(argv[event.pid])):
             skip = True
 
         if not skip:
             if args.timestamp:
                 print("%-8.3f" % (time.time() - start_ts), end="")
             ppid = get_ppid(event.pid)
-            print("%-16s %-6s %-6s %3s %s" % (event.comm.decode(), event.pid,
-                    ppid if ppid > 0 else "?", event.retval,
-                    b' '.join(argv[event.pid]).decode()))
+            ppid = b"%d" % ppid if ppid > 0 else b"?"
+            printb(b"%-16s %-6d %-6s %3d %s" % (event.comm, event.pid,
+                   ppid, event.retval, b' '.join(argv[event.pid])))
```

Comment:
Is there a way to do this without changing print() to printb()? I can get my head around kernel strings needing b"", but this is affecting non-kernel strings too (like printing an int, or b"?"). Something like this would seem easier:

```Python
            print("%-16s %-6d %-6s %3d %s" % (b(event.comm), event.pid,
                   ppid, event.retval, b' '.join(argv[event.pid])))
```

We only special case the kernel strings.

Plus, since these are pulled in via:

```Python
 class Data(ct.Structure):
     _fields_ = [
         ("pid", ct.c_uint),
         ("comm", ct.c_char * TASK_COMM_LEN),
         ("type", ct.c_int),
         ("argv", ct.c_char * ARGSIZE),
         ("retval", ct.c_int),
]
```

It makes me wonder if we can deal with it there, and leave the entire print() call unchanged. eg, something like:

```Python
 class Data(ct.Structure):
     _fields_ = [
         ("pid", ct.c_uint),
         ("comm", b.k_char * TASK_COMM_LEN),
         ("type", ct.c_int),
         ("argv", b.k_char * ARGSIZE),
         ("retval", ct.c_int),
]
```

b.k_char == kernel character.